tsl/src/continuous_aggs/common.c

```diff
@@ -891,6 +932,25 @@ cagg_validate_query(const Query *query, const bool finalized, const char *cagg_s
 				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 				 errmsg("cannot create continuous aggregate on hypertable with row security")));
 
+	/* Test for broken time_bucket configurations (variable with with offset and origin). We need to
+	 * check only time based buckets since integer based bucks are always fixed. */
+	bool time_offset_or_origin_set =
+		(bucket_info.bucket_time_offset != NULL) ||
+		(TIMESTAMP_NOT_FINITE(bucket_info.bucket_time_origin) == false);
+
+	/* Ignore time_bucket_ng in this check, since offset and origin was allowed in the past */
+	FuncInfo *func_info = ts_func_cache_get_bucketing_func(bucket_info.bucket_func->funcid);
+	bool is_time_bucket_ng = func_info->origin == ORIGIN_TIMESCALE_EXPERIMENTAL;
+
+	if (time_bucket_info_has_fixed_width(&bucket_info) == false && time_offset_or_origin_set &&
+		!is_time_bucket_ng)
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("cannot create continuous aggregate with variable-width bucket using "
+						"offset or origin.")));
+	}
```

Comment:
This does not seem like a design issue but rather about a limitation in the implementation. Can you add a comment describing the situation so that we eventually fix it?

Reply:
Comment is added as suggested